Automattic/wp-calypso
# Help center: Add fallback and tracking event when API fails to load the article

Slack: p1722917613524509-slack-C06DN6QQVAQ

## Proposed Changes

Add a Fallback to the Help center when the article is not loaded by the API and a tracking event when it fails.

[Screencast from 2024-08-06 18-51-25.webm](https://github.com/user-attachments/assets/4a59a43f-a55a-4197-aef7-9d5498d219ed)

## Why are these changes being made?
* Help center keeps showing as it's loading.

## Testing Instructions

As the API error mentioned here (p1722917613524509-slack-C06DN6QQVAQ) was fixed, we have to make it fail.

* Edit the link with an invalid one [here](https://github.com/Automattic/wp-calypso/blob/a642be32a51e5e9fdcecbea08d0783cb980b2fd8/client/components/inline-support-link/context-links.js#L344) 
* Go to `/github-deployments/:YOUR_SITE`
* Click on `Learn More` in the `Deployments` subtitle.
* You should see an error message
* Check the tracking event `calypso_helpcenter_article_api_error`
* Click on the `Click here` link in the message. It should link to the original link (modified in step 1)

## Pre-merge Checklist
